### PR TITLE
Reimplement AWS trust service methods after merging #92

### LINKED ISSUE #93: Use generic trust service method in the trust_ec2_resources method
A generic trust method was introduced in #92.

### PR DESCRIPTION
**Pull request by erran-r7**
Closes #93.

---

~~**NOTE**: These changes will update the names for most of these policies.~~

*edit*: Changes to maintain the policy names were implemented.

### CODE REVIEW

**Comment by simonirwin-r7** [comment]
@erran-r7 I don't *think* the policy name (which will change as part of this PR) actually gets used.

Here's a snippet of a CF template generated by Convection:

```
"AssumeRolePolicyDocument": {
          "Version": "2012-10-17",
          "Statement": [
            {
              "Effect": "Allow",
              "Action": [
                "sts:AssumeRole"
              ],
              "Principal": {
                "Service": "ec2.amazonaws.com"
              }
            }
          ]
        }
```

**Comment by athompson-r7** [comment]
I don't like the idea of the name changing.  Anything we can do to avoid this?

**Comment by erran-r7** [comment]
@athompson-r7 Yes, simply add a name parameter to the generic message that isn't required. It'll default to the `"trust-#{service}-service"` name format (new) but we can pass in the old names to ensure consistency when using these methods.

**Comment by erran-r7** [comment]
> I don't like the idea of the name changing. Anything we can do to avoid this?



> @athompson-r7 Yes, simply add a name parameter to the generic message that isn't required. It'll default to the "trust-#{service}-service" name format (new) but we can pass in the old names to ensure consistency when using these methods.

@athompson-r7 check out 3932f60 where we keep parity with the existing methods by maintaining the policy names.

File: lib/convection/model/template/resource/aws_iam_role.rb [hunk 1]
```diff
@@ -36,49 +37,23 @@ def trust_service(name, &block)
...
           def trust_emr(&block)
-            @trust_relationship = Model::Mixin::Policy.new(:name => 'trust-emr', :template => @template)
-            trust_relationship.allow do
-              action 'sts:AssumeRole'
-              principal :Service => 'elasticmapreduce.amazonaws.com'
-            end
-            trust_relationship.instance_exec(&block) if block
-            trust_relationship
+            trust_service('elasticmapreduce', 'trust-emr', &block)
+            trust_service('', '', &block)
```
Comment (erran-r7) [inline lib/convection/model/template/resource/aws_iam_role.rb]: eek! 🍝

Comment (erran-r7) [inline lib/convection/model/template/resource/aws_iam_role.rb]: Fixed in https://github.com/rapid7/convection/pull/94/commits/d857719fbeff7a58ed16b95c2616f9d3145f937a.

**Pull request merged**